File: src/models/branch.ts
export interface Branch {
  name: string
  merged: boolean
  protected: boolean
  default: boolean
  can_push: boolean
  web_url: string
  commit: Commit
}

export interface Commit {
  id: string
  author_name: string
  committer_name: string
  committed_date: string
  title: string
  message: string
}

File: api/src/main/kotlin/com/github/larscom/gitlabcidashboard/branch/model/Branch.kt
package com.github.larscom.gitlabcidashboard.branch.model

import com.fasterxml.jackson.annotation.JsonAlias
import java.time.Instant

data class Branch(
    val name: String,
    val merged: Boolean,
    val protected: Boolean,
    val default: Boolean,
    @JsonAlias("can_push") val canPush: Boolean,
    @JsonAlias("web_url") val webUrl: String,
    val commit: Commit
) {
    data class Commit(
        val id: String,
        @JsonAlias("author_name")
        val authorName: String,
        @JsonAlias("committer_name")
        val committerName: String,
        @JsonAlias("committed_date")
        val committedDate: Instant,
        val title: String,
        val message: String
    )
}
